rbd/pool_nautilus_test.go

```diff
@@ -131,3 +132,80 @@ func TestPoolInit(t *testing.T) {
 		assert.NoError(t, err)
 	})
 }
+
+func TestGetAllPoolStat(t *testing.T) {
+	conn := radosConnect(t)
+	poolName := GetUUID()
+	err := conn.MakePool(poolName)
+	assert.NoError(t, err)
+
+	ioctx, err := conn.OpenIOContext(poolName)
+	assert.NoError(t, err)
+	assert.NotNil(t, ioctx)
+
+	defer func() {
+		ioctx.Destroy()
+		conn.DeletePool(poolName)
+		conn.Shutdown()
+	}()
+
+	poolstats := poolStatsCreate()
+	defer func() {
+		poolstats.destroy()
+	}()
+
+	var imageName string
+	size := uint64(2 << 20)
+	var expectedSize uint64
+
+	for idx := 0; idx < 3; idx++ {
+		imageName = GetUUID()
+		image, err := Create(ioctx, imageName, size, testImageOrder)
+		assert.NoError(t, err)
+		defer func() {
+			assert.NoError(t, image.Remove())
+		}()
+		expectedSize += size
+	}
+
+	imageName = GetUUID()
+	_, err = Create(ioctx, imageName, size, testImageOrder)
+	assert.NoError(t, err)
+	expectedSize += size
+	img, err := OpenImage(ioctx, imageName, NoSnapshot)
+	assert.NoError(t, err)
+
+	mySnap, err := img.CreateSnapshot("mySnap")
+	assert.NoError(t, err)
+
+	t.Run("NullIOContext", func(t *testing.T) {
+		_, err = GetAllPoolStats(nil)
+		assert.Error(t, err)
+	})
+
+	t.Run("CheckPoolStatOption", func(t *testing.T) {
+		err := img.Resize(size)
+		assert.NoError(t, err)
+		assert.NoError(t, mySnap.Remove())
+		assert.NoError(t, img.Close())
+		err = img.Trash(time.Hour)
+		assert.NoError(t, err)
+		defer func() {
+			trashList, err := GetTrashList(ioctx)
+			assert.NoError(t, err)
+			assert.NoError(t, TrashRemove(ioctx, trashList[0].Id, true))
+		}()
+
+		omap, err := GetAllPoolStats(ioctx)
+		assert.NoError(t, err)
+
+		assert.Equal(t, uint64(3), omap[PoolStatOptionImages])
+		assert.Equal(t, (expectedSize - size), omap[PoolStatOptionImageProvisionedBytes])
+		assert.Equal(t, expectedSize-size, omap[PoolStatOptionImageMaxProvisionedBytes])
```

Comment:
No biggie, by why the parens on the line above and none here?

Reply:
Not intentional, just an accident.